src/main/components/AssertionsForm.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useContext } from 'react';
+import styled from 'styled-components';
+import { TestsContext } from '../testsContext';
+import Select from './InlineSelect.jsx';
+import InlineInput from './InlineInput.jsx';
+import Form from './InlineForm.jsx';
+import Button from './Button.jsx';
+
+// Styles
+const FormWrapper = styled.section`
+  background: #F0F3F4;
+  border-radius: 5px;
+  box-shadow: inset 0 0 2px 0 rgba(41, 47, 50, 0.25);
+  padding: 0.5em 1.5em;
+  h3 {
+    color: #555B5E;
+    font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+    font-feature-settings: 'ss18';
+    font-size: 1em;
+    letter-spacing: 0.075em;
+    text-transform: uppercase;
+  }
+`;
+
+const Input = styled(InlineInput)`
+  background: none;
+  margin: 0 1em;
+  padding: 0 0 0.25em 0.25em;
+  text-align: center;
+`;
+
+const Fieldset = styled.fieldset`
+  border: none;
+`;
+
+const Label = styled.label`
+  color: #292F32;
+  font-family: 'Halyard Text', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
+`;
+
+const AssertionsForm = (props) => {
+  const { index, setMode } = props;
+
+  // State
+  const [isRange, setIsRange] = useState(true);
+  const [minCode, setMinCode] = useState(200);
+  const [maxCode, setMaxCode] = useState(299);
+  const [exactCode, setExactCode] = useState(200);
+  const [tests, setTests] = useContext(TestsContext);
+
+  // Event handlers
+  const handleChange = (event) => {
+    const { value, name } = event.target;
+    switch (name) {
+      case 'rangeSelector':
+        if (value === 'range') setIsRange(true);
+        else setIsRange(false);
+        break;
+      case 'min-code':
+        setMinCode(Number.parseInt(value, 10));
+        break;
+      case 'max-code':
+        setMaxCode(Number.parseInt(value, 10));
+        break;
+      case 'exact-code':
+        setExactCode(Number.parseInt(value, 10));
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleSubmit = (event) => {
```

Comment:
I'm a bit confused as to how this works. I understand that it sets specific values based on user input, but what I don't understand is what are those values and what are they used for? When we get a better understanding of the code base, we'll have to add a lot of comments (So far comments have been far and few between).

Reply:
🤔